Added the ability to limit the size of a collection

dpobel opened the pull request:
As the title says, this patch adds the ability to limit the size of a collection by using a `limit` option in its definition.

My use case: in the home page of my website, I want to display the 10 last blog posts (and I don't want to write a template filter/function/operator ;-))

dominicbarnes commented:
I think this type of feature has some merit, so I will work on getting it merged in. @dpobel if you have the time/interest to refactor the branch, it would help me a lot.

dpobel commented:
great news! I'm still interested as I'm using my custom branch in my projects.
I rebased the branch and fixed the conflict. The unit tests are passing. I hope that's ok.
